src/exo/new_eff.py

```diff
@@ -2121,7 +2121,7 @@ def Check_ExprBound(proc, stmts, expr, op, value, exception=True):
         query = ADef(e < AInt(value))
         err_msg = f"greater than {value}"
     elif op == "==":
-        query = ADef(e=AInt(value))
+        query = AEq(e, AInt(value))
```

Comment:
What's the difference between querying `ADef(expr)` and just querying `expr`? I notice all the other cases wrap their queries with a `ADef`, and I'm wondering if `ADef(AEq(e, AInt(value)))` is different from just `AEq(e, AInt(value))`

Reply:
Good catch, I am actually not sure. I will query `ADef(AEq(e, AInt(value)))` just for consistency. Should be fixed now.

Reply:
@yamaguchi1024 do you know?

Reply:
IIUC, having `ADef` around will ensure that the statement is definitely true (i.e. the reachability of location sets is also checked). If you don't have `ADef`, it would mean "there **may** be some input values where this statement is true".